packages/contracts/src/core/dao/DAO.sol

```diff
@@ -122,6 +154,21 @@ contract DAO is
         __PermissionManager_init(_initialOwner);
     }
 
+    /// @notice Initializes the DAO after an upgrade from a previous version.
+    /// @param previousProtocolVersion The protocol version number of the previous DAO implementation contract this upgrade is transitioning from.
+    function initializeUpgradeFrom(
```

Comment:
Note that In a future update, this function might get more arguments or a generic `bytes calldata data` argument that it can `abi.decode` to have the initialization data that it needs.

Reply:
See https://github.com/aragon/simple-storage-example-plugin/blob/6df1fc6216ad5ff818c497d0a429b026927facc3/contracts/release1/build3/SimpleStorageR1B3.sol#L42-L50, where I have explored this in the `SimpleStorage` example plugin.

Reply:
@Michael-A-Heuer shouldn't we have the generic `bytes memory` now? It might be more convenient to have these init functions with a static and predictable ABI that doesn't change over time.

Also, we might want to define them in a new interface?

Reply:
> shouldn't we have the generic bytes memory now? It might be more convenient to have these init functions with a static and predictable ABI that doesn't change over time.

I saw no need for the contracts but from an SDK viewpoint, I guess it makes sense to add it now already.

> Also, we might want to define them in a new interface?

I think it would be wrong to put initializers in an interface. Like the constructor, they belong tightly to the contract that they initialize and are intentionally left out the interface, as they can be expected to change.
I would not create an interface for `initializeUpgradeFrom(uint8[3] _protocolVersion, bytes calldata _initData)`

Reply:
Added the `bytes calldata` parameter in cf3b2c1fad96c0d4d7c75d5f03c2c06ec8a35528

Reply:
what's the best function name btw?
- `initializeUpgradeFrom`, 
- `initializeFromUpgrade` or just 
- `initializeFrom`?

Reply:
I personally prefer `initializeFrom([1, 3, 0], bytes(""))`

Reply:
About the interface, it's not that plugins need to have it. But rather an interface that each plugin chooses to adopt. It will help the PluginSetup know what to call

Reply:
> I personally prefer `initializeFrom([1, 3, 0], bytes(""))`

Same, changed in 58a9250d77aa96085b997e364f8ad72035b87918

> About the interface, it's not that plugins need to have it. But rather an interface that each plugin chooses to adopt. It will help the PluginSetup know what to call

For the `PluginSetup` it is a different interface anyway:
`initializeFrom(uint16 _build, bytes calldata _initData)`
I would not create an interface for the initializers - these functions are only called once and only because of EVM limitations regarding upgradeable contracts - otherwise, we would use a constructors.

Reply:
We won't create an interface. All other changes were applied. Thus, we can resolve this